### PR TITLE
Feature/cc 0068 user can change color of the court

### PR DESCRIPTION
**Pull request by EmilyGong111**
A user click a color on the color board，then click different court area, the court area's color gonna change to be the color the use just chosed from the color board.

For example, I clicked on the light blue color in the color board.
Then I clicked on the court border, the border color changed to light blue.

Screen Shot 2022-06-27 at 11 10 22 pm
Screen Shot 2022-06-27 at 11 11 47 pm

### CODE REVIEW

File: src/components/BasketballCourt/Border.tsx [hunk 1]
```diff
@@ -10,17 +11,25 @@ const Border: React.FC<BorderProps> = ({ startPoint }) => {
     (state) => state.courtSize
   );
 
-  const color = useStoreSelector(
-    (state) => state.tile.find((tile) => tile.location.includes("border"))?.color
-  );
+  // const color = useStoreSelector(
+  //   (state) => state.tile.find((tile) => tile.location.includes("border"))?.color
+  // );
+
```
Comment (Jzeva) [inline src/components/BasketballCourt/Border.tsx:17]: Unused code can be deleted before submitting.

File: src/components/BasketballCourt/CircleArea.tsx [hunk 2]
```diff
@@ -10,9 +11,15 @@ const CircleArea: React.FC<CircleAreaProps> = ({ startPoint }) => {
   const { courtAreaXLength, threePointLineToCourtEdgeLenth, threePointLineRadius, circleRadius } =
     useStoreSelector((state) => state.courtSize);
 
-  const color = useStoreSelector(
-    (state) => state.tile.find((tile) => tile.location.includes("circleArea"))?.color
-  );
+  // const color = useStoreSelector(
+  //   (state) => state.tile.find((tile) => tile.location.includes("circleArea"))?.color
+  // );
```
Comment (Jzeva) [inline src/components/BasketballCourt/CircleArea.tsx:16]: Same here

File: src/components/BasketballCourt/CourtArea.tsx [hunk 3]
```diff
@@ -11,19 +12,25 @@ interface CourtAreaProps {
 const CourtArea: React.FC<CourtAreaProps> = ({ courtWidth, startPoint }) => {
   const { courtAreaYLength } = useStoreSelector((state) => state.courtSize);
 
-  const color = useStoreSelector(
-    (state) => state.tile.find((tile) => tile.location.includes("courtArea"))?.color
-  );
+  // const color = useStoreSelector(
+  //   (state) => state.tile.find((tile) => tile.location.includes("courtArea"))?.color
+  // );
```
Comment (Jzeva) [inline src/components/BasketballCourt/CourtArea.tsx:17]: Same here

**Review by Jzeva — APPROVED** [review]
LGTM apart from those unused code. I also leaned a lot from your tests. Thanks for that.

**Comment by EmilyGong111** [comment]
> LGTM apart from those unused code. I also leaned a lot from your tests. Thanks for that. BTW, don't forget to fix the conflict. I have checked that conflict. It should just be some space problem which is pretty easier to fix.

Thanks!

File: src/components/BasketballCourt/Border.tsx [hunk 4]
```diff
@@ -9,18 +10,21 @@ const Border: React.FC<BorderProps> = ({ startPoint }) => {
   const { courtAreaXLength, courtAreaYLength, borderLength } = useStoreSelector(
     (state) => state.courtSize
   );
+  const currentColorBoardColor = useStoreSelector((state) => state.courtColor.color);
```
Comment (tatoMa) [inline src/components/BasketballCourt/Border.tsx:13]: I think "currentColorBoardColor" is not a proper name for this value.

File: src/components/BasketballCourt/Border.tsx [hunk 5]
```diff
@@ -9,18 +10,21 @@ const Border: React.FC<BorderProps> = ({ startPoint }) => {
...
+  const [courtBorderColor, setCourtAreaColor] = useState("#195955");
```
Comment (tatoMa) [inline src/components/BasketballCourt/Border.tsx:14]: This naming convention may not good for hooks. Normally people use [something, useSomething] define useState. These 2 destructured names should be the same apart from "use".

File: src/components/BasketballCourt/Border.tsx (hunk 4 above)
Comment (EmilyGong111) [reply src/components/BasketballCourt/Border.tsx]: Thanks, I changed the name to be “selectedColor".

File: src/components/BasketballCourt/ThreePointArea.tsx [hunk 6]
```diff
@@ -21,10 +22,12 @@ const ThreePointArea: React.FC<ThreePointAreaProps> = ({ startPoint }) => {
   const controlPointFourX = startPoint.X + cornerThreePointLineLength;
   const controlPointFourY = controlPointThreeY;
 
-  const color = useStoreSelector(
-    (state) => state.tile.find((tile) => tile.location.includes("threePoint"))?.color
-  );
+  const selectedColor = useStoreSelector((state) => state.courtColor.color);
+  const [courtThreePointColor, setThreePointAreaColor] = useState("#72818B");
```
Comment (tatoMa) [inline src/components/BasketballCourt/ThreePointArea.tsx:26]: Naming convention issue.

**Review by tatoMa — COMMENTED** [review]
Well done. Now LGTM.
Please also raise another ticket about fixing the existing code.

KeyArea.ts #7 the name of the interface, delete "text".
TopKeyArea.ts #7 the name of the interface change it to the one that fit the file name.

**Review by tatoMa — APPROVED** [review]

**Comment by EmilyGong111** [comment]
> 

Thanks!
It's Done.

File: src/components/BasketballCourt/KeyArea.tsx [hunk 7]
```diff
@@ -2,28 +2,32 @@ import { Rect } from "react-konva";
...
     <Rect
       width={keyAreaWidth}
       height={keyAreaHeight}
-      fill={color}
+      fill={keyAreaColor}
       stroke="white"
       strokeWidth={courtWhiteLine}
       x={startPoint.X}
       y={startPoint.Y + threePointLineToCourtEdgeLenth + threePointLineRadius - keyAreaHeight / 2}
+      onClick={() => handleColorChange()}
```
Comment (Fuchih) [inline src/components/BasketballCourt/KeyArea.tsx:30]: if you won't pass any  props, you can simples use "onClick={handleColorChange}" 
you don't need need to change it,  just  a remider

**Review by Fuchih — APPROVED** [review]

File: src/components/BasketballCourt/KeyArea.tsx (hunk 7 above)
Comment (EmilyGong111) [reply src/components/BasketballCourt/KeyArea.tsx:30]: Thanks for the reminder!

**Pull request merged**